LoopKit/InsulinKit/PersistedOutsideDoseEvent.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  PersistedOutsideDoseEvent.swift
+//  LoopKit
+//
+//  Created by Anna Quinlan on 4/17/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+import Foundation
+
+public struct PersistedOutsideDoseEvent {
```

Comment:
Would prefer that we use existing types (i.e. primarily CachedInsulinDeliveryObject) to track outside doses. Is there some reason that this doesn't work?